ext/RMagick/rmpixel.c

```diff
@@ -236,10 +223,86 @@ Pixel_alpha_eq(VALUE self, VALUE v)
 /*
  * Get/set Pixel CMYK attributes.
  */
-DEF_PIXEL_CMYK_CHANNEL_ACCESSOR(cyan, red)
-DEF_PIXEL_CMYK_CHANNEL_ACCESSOR(magenta, green)
-DEF_PIXEL_CMYK_CHANNEL_ACCESSOR(yellow, blue)
-DEF_PIXEL_CMYK_CHANNEL_ACCESSOR(black, black)
+VALUE
+Pixel_cyan(VALUE self)
+{
+    Pixel *pixel;
+
+    Data_Get_Struct(self, Pixel, pixel);
+    return INT2NUM(pixel->red);
+}
+VALUE
+Pixel_cyan_eq(VALUE self, VALUE v)
+{
+    Pixel *pixel;
+
+    rb_check_frozen(self);
+    Data_Get_Struct(self, Pixel, pixel);
+    pixel->red = APP2QUANTUM(v);
+    (void) rb_funcall(self, rm_ID_changed, 0);
+    (void) rb_funcall(self, rm_ID_notify_observers, 1, self);
+    return QUANTUM2NUM(pixel->red);
```

Comment:
@Watson1978 could these be accessors in ruby instead?

```rb
module Magick
  class Pixel
    attr_accessor :cyan
  end
end
```

Or does it need to be in C so that we can access it from C code?

Reply:
There is no particular reason to define it in C language.